packages/react-server/src/ReactFizzServer.js

```diff
@@ -1652,6 +1653,7 @@ function flushSubtree(
       // We're emitting a placeholder for this segment to be filled in later.
       // Therefore we'll need to assign it an ID - to refer to it by.
       const segmentID = (segment.id = request.nextSegmentId++);
+      segment.textEmbedding = textEmbeddingForDelayedSegment();
```

Comment:
Is it always the case that nothing has been written to this segment yet? I think maybe yes, but not sure.

Reply:
If some extra separators come along for the ride I don't think it should matter. As long as we are using the patchup in DOM we can handle separators but don't need them. so in this path we know we are getting a placeholder and thus will also get a patchup. Changing the embedding here just means that we have an opportunity to drop them if they have not yet been emitted.

That said... I do not know how it could have had anything written without being complete and thus avoiding this path

Reply:
and actually for delayed segments we don't want to emit a final separator but they aren't boundaries so I need a way to know if it is delayed or not when I get to the final push

Reply:
but I can just track that all as another property on the Segment